catalog/app/utils/MetaTitle.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react'
+import { Helmet } from 'react-helmet'
+
+const BASE = 'Quilt is a versioned data portal for AWS'
+
+const DIVIDER = ' • '
+
+interface MetaTitleProps {
+  children?: string | string[]
+  base?: string
+}
+
+function getTitleSegments(base: string, children?: string | string[]) {
+  if (!children) return [base]
+
+  if (Array.isArray(children)) return [...children, base]
+
+  return [children, base]
+}
+
+export function getTitle(base: string, children?: string | string[]) {
+  return getTitleSegments(base, children).join(DIVIDER)
+}
+
+export default function MetaTitle({ children, base = BASE }: MetaTitleProps) {
+  const titleStr = React.useMemo(() => getTitle(base, children), [base, children])
```

Comment:
this will be re-evaluated every render, since the most common use case is smth like `<MetaTitle>{[a, b]}</MetaTitle>` -- children array is created anew each render and useMemo wont work